### ISSUE TITLE
Task ids not shown inside task group

### ISSUE THREAD

**Issue by dstandish** [apache/airflow#33093]
### Body

task ids not shown in graph.
note the tasks are not actually setup / teardown just normal tasks named that way.

<img width="1173" alt="image" src="https://github.com/apache/airflow/assets/15932138/d31a8fb9-df2e-472b-a743-620511c472c0">

```python
with DAG(dag_id="leaves_ignored", start_date=pendulum.now()):
    @task
    def setup1():
        ...
    @task
    def setup2():
        ...
    @task
    def teardown1():
        ...
    @task
    def teardown2():
        ...
    @task
    def work1():
        ...
    @task
    def work2():
        ...


    setup1 = setup1()
    # setup2 = setup2()
    teardown1 = teardown1()
    # teardown2 = teardown2()
    work1 = work1()
    work2 = work2()

    with TaskGroup("my_group") as tg:
        setup1 >> work1 >> teardown1
        setup1 >> teardown1
    tg >> work2
```

@bbovenzi 

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

**Comment by dstandish** [comment]
Update: we must now use `add_task` to add the tasks to group context:
```python
    with TaskGroup("my_group") as tg:
        setup1 >> work1 >> teardown1
        setup1 >> teardown1
        tg.add_task(setup1)
        tg.add_task(teardown1)
        tg.add_task(work1)
    tg >> work2
```

**Comment by eladkal** [comment]
> we must now use add_task to add the tasks to group context

Did https://github.com/apache/airflow/pull/33262 solve this issue?

**Comment by eladkal** [comment]
Checked with @ephraimbuddy this was solved in 2.7.0rc2
